### PR TITLE
ci: release 0.4.0

### PR DESCRIPTION
**Pull request by langchain-infra**

### CODE REVIEW

File: charts/langsmith/README.md [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
-There are a few important changes when migrating from 0.3.0 to 0.4.0:
+There are a few important changes when migrating from 0.3.0 to 0.4.0. The majority of these will require no action on your part. However, there are a few things to note:
 
 - OAuth Flow now relies on using Access Tokens instead of OIDC ID tokens. This shouldn't impact any of your application functionality.
+- A new asynchronous queue worker implementation has been added. This should improve the performance of the application trace ingestion.
 - Clickhouse persistence now uses 50Gi of storage by default. You can adjust this by changing the `clickhouse.statefulSet.persistence.size` value in your `values.yaml` file.
   - You may need to resize your existing storage class or set `clickhouse.statefulSet.persistence.size` to the old default value of `8Gi`.
 - Some our image repositories have been update. You can see the root repositories in our `values.yaml` file. You may need to update mirrors.
+- We now expose an api key salt parameter. This previously defaulted to your LangSmith License Key. You should set this param to your license key to avoid invalidating old api keys.
```
Comment (nfcampos) [inline charts/langsmith/README.md:18]: I think you mean "you should not set..."

File: charts/langsmith/README.md [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 
 ## Migrating from LangSmith 0.3.0 to 0.4.0
 
 LangSmith 0.3.0 should be a drop-in replacement for LangSmith 0.2.0. You can follow the generic upgrade instructions [here](docs/UPGRADE.md).
 
-There are a few important changes when migrating from 0.3.0 to 0.4.0:
+There are a few important changes when migrating from 0.3.0 to 0.4.0. The majority of these will require no action on your part. However, there are a few things to note:
 
 - OAuth Flow now relies on using Access Tokens instead of OIDC ID tokens. This shouldn't impact any of your application functionality.
+- A new asynchronous queue worker implementation has been added. This should improve the performance of the application trace ingestion.
```
Comment (nfcampos) [inline charts/langsmith/README.md:14]: this should say sth like "replacing previous synchronous queue worker" to make clear this isn't an additional component

File: charts/langsmith/README.md [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
 - Clickhouse persistence now uses 50Gi of storage by default. You can adjust this by changing the `clickhouse.statefulSet.persistence.size` value in your `values.yaml` file.
   - You may need to resize your existing storage class or set `clickhouse.statefulSet.persistence.size` to the old default value of `8Gi`.
 - Some our image repositories have been update. You can see the root repositories in our `values.yaml` file. You may need to update mirrors.
```
Comment (nfcampos) [inline charts/langsmith/README.md:17]: "have been updated"

File: charts/langsmith/README.md [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
+- We now expose an api key salt parameter. This previously defaulted to your LangSmith License Key. You should set this param to your license key to avoid invalidating old api keys.
+
+** Note: Using a new api key salt will invalidate all old api keys. **
```
Comment (nfcampos) [inline charts/langsmith/README.md:20]: which makes migrating to this a bit hard? should we suggest people set their *current* license as the api key salt env var and use that forever going fwd?

Comment (langchain-infra) [reply charts/langsmith/README.md:20]: Right, thats why I added the "should set this param to your license key" but maybe should clarify only if migrating

**Pull request merged**